.github/workflows/iwyu.yml

```diff
@@ -111,22 +115,32 @@ jobs:
         run: |
           PWD=$(pwd)
           # -isystem/usr/lib/clang/17/include
-          iwyu_tool -p cmake.output -j $(nproc) -- -w -Xiwyu --max_line_length=1024 -Xiwyu --comment_style=long -Xiwyu --quoted_includes_first -Xiwyu --update_comments > iwyu.log
+          iwyu_tool -p cmake.output -j $(nproc) -- -w -Xiwyu --max_line_length=1024 -Xiwyu --comment_style=long -Xiwyu --quoted_includes_first -Xiwyu --update_comments -stdlib=${{ matrix.stdlib }} > iwyu.log
```

Comment:
This should not be necessary since it already is specified in the compilation database:

```json
{
  "directory": "/__w/cppcheck/cppcheck/cmake.output/externals/tinyxml2",
  "command": "/usr/bin/clang++ -DCHECK_INTERNAL -DFILESDIR=\\\"/usr/local/share/Cppcheck\\\" -DHAVE_RULES -I/__w/cppcheck/cppcheck/cmake.output/externals/tinyxml2 -I/__w/cppcheck/cppcheck/externals/tinyxml2 -O3 -DNDEBUG -std=gnu++11 -Weverything -O2 -pedantic -Wall -Wextra -Wcast-qual -Wfloat-equal -Wmissing-declarations -Wmissing-format-attribute -Wno-long-long -Wpacked -Wredundant-decls -Wundef -Wno-missing-field-initializers -Wno-missing-braces -Wno-sign-compare -Wno-multichar -Woverloaded-virtual -mllvm -inline-deferral -gdwarf-4 -stdlib=libc++ -Wno-documentation-unknown-command -Wno-unused-exception-parameter -Wno-old-style-cast -Wno-sign-conversion -Wno-shadow-field-in-constructor -Wno-covered-switch-default -Wno-shorten-64-to-32 -Wno-implicit-int-conversion -Wno-double-promotion -Wno-shadow-field -Wno-shadow-uncaptured-local -Wno-implicit-float-conversion -Wno-switch-enum -Wno-float-conversion -Wno-enum-enum-conversion -Wno-date-time -Wno-disabled-macro-expansion -Wno-bitwise-instead-of-logical -Wno-switch-default -Wno-unsafe-buffer-usage -Wno-global-constructors -Wno-exit-time-destructors -Wno-padded -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-four-char-constants -Wno-weak-vtables -U_GLIBCXX_DEBUG -Wno-implicit-fallthrough -Wno-suggest-destructor-override -Wno-zero-as-null-pointer-constant -Wno-format-nonliteral -Wno-inconsistent-missing-destructor-override -o CMakeFiles/tinyxml2_objs.dir/tinyxml2.cpp.o -c /__w/cppcheck/cppcheck/externals/tinyxml2/tinyxml2.cpp",
  "file": "/__w/cppcheck/cppcheck/externals/tinyxml2/tinyxml2.cpp",
  "output": "externals/tinyxml2/CMakeFiles/tinyxml2_objs.dir/tinyxml2.cpp.o"
}
```

I filed https://github.com/include-what-you-use/include-what-you-use/issues/1605 upstream about it.